tests/ops/test_stats.py

```diff
@@ -1,12 +1,90 @@
 import pytest
 import torch
 
-from pyro.ops.stats import (autocorrelation, autocovariance, effective_sample_size,
-                            gelman_rubin, split_gelman_rubin,
-                            _cummin, _fft_next_good_size)
+from pyro.ops.stats import (autocorrelation, autocovariance, effective_sample_size, gelman_rubin,
+                            hpdi, pi, quantile, resample, split_gelman_rubin, waic, _cummin,
+                            _fft_next_good_size)
 from tests.common import assert_equal, xfail_if_not_implemented
 
 
+@pytest.mark.parametrize('replacement', [True, False])
+@pytest.mark.init(rng_seed=3)
+def test_resample(replacement):
+    x = torch.empty(10000, 2)
+    x[:, 0].normal_(3, 4)
+    x[:, 1].normal_(5, 6)
+
+    num_samples = 5000
+    y = resample(x, num_samples=num_samples, replacement=replacement)
+    z = resample(x.t(), num_samples=num_samples, dim=1, replacement=replacement)
+    if not replacement:
+        assert_equal(torch.unique(y.reshape(-1)).numel(), y.numel())
+        assert_equal(torch.unique(z.reshape(-1)).numel(), z.numel())
+    assert_equal(y.shape, torch.Size([num_samples, 2]))
+    assert_equal(z.shape, torch.Size([2, num_samples]))
+    assert_equal(y.mean(dim=0), torch.tensor([3., 5.]), prec=0.1)
```

Comment:
Could you also run these tests on CUDA and confirm that they pass?

Reply:
@neerajprad It is sad that I don't have a GPU machine at my hand.

Reply:
Np at all. Let me test and get back on this.

Reply:
We might have to bump this to 0.2 for CUDA.